lib/FieldLocation/FieldLocationFinal.js

```diff
@@ -1,103 +1,86 @@
 
 import { stripesConnect } from '@folio/stripes/core';
 import { LocationLookup } from '@folio/stripes/smart-components';
 import {
-  IconButton,
-  TextField,
+  Loading,
 } from '@folio/stripes/components';
 
+import { FieldSelectFinal } from '../FieldSelect';
 import { locationByPropManifest } from '../manifests';
-import { validateRequired } from '../utils';
+import { validateRequired, getLocationOptions } from '../utils';
+import { fieldSelectOptionsShape } from '../shapes';
 
 const FieldLocationFinal = ({
-  locationId,
   labelId,
+  locationId,
+  locationLookupLabel,
+  locationOptions,
   mutator,
   name,
   onChange,
   required,
 }) => {
-  const [selectedLocation, setSelectedLocation] = useState({});
+  const [locationsForSelect, setLocationsForSelect] = useState();
+  const isLocationFromOptions = Boolean(find(locationOptions, { 'value': locationId }));
 
   useEffect(() => {
-    if (locationId && selectedLocation.id !== locationId) {
+    setLocationsForSelect(locationOptions);
+
+    if (locationId && !isLocationFromOptions) {
       mutator.fieldLocation.GET()
-        .then(setSelectedLocation);
+        .then(locationResponse => setLocationsForSelect(prev => [...prev, ...getLocationOptions([locationResponse])]));
     }
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [locationId]);
+  }, [locationId, locationOptions, isLocationFromOptions]);
 
   const selectLocation = useCallback(
     (location) => {
-      setSelectedLocation(location);
+      setLocationsForSelect(prev => uniqBy([...prev, ...getLocationOptions([location])], 'value'));
 
       onChange(location);
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [onChange],
   );
 
-  const clearLocation = useCallback(
-    () => {
-      setSelectedLocation({});
-
-      onChange({});
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [onChange],
-  );
-
-  const clearButton = useMemo(
-    () => {
-      if (selectedLocation.id) {
-        return (
-          <IconButton
-            icon="times-circle-solid"
-            onClick={clearLocation}
-            size="small"
-          />
-        );
-      }
-
-      return null;
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [selectedLocation],
-  );
+  if (!locationsForSelect) return <Loading />;
 
   return (
     <div>
-      <Field
-        component={TextField}
-        disabled
-        endControl={clearButton}
-        format={() => selectedLocation.name}
+      <FieldSelectFinal
+        dataOptions={locationsForSelect}
         fullWidth
-        hasClearIcon={false}
         id={name}
         label={labelId ? <FormattedMessage id={labelId} /> : ''}
         marginBottom0
         name={name}
         required={required}
         validate={required ? validateRequired : undefined}
       />
-      <LocationLookup onLocationSelected={selectLocation} />
+      <LocationLookup
+        label={locationLookupLabel}
```

Comment:
what was the previous label, is it required to be changed?

Reply:
Previous was the default - `Location lookup`, but now it should be `Assign a different location`